src/app/customizer/gradient-generator/page.js

```diff
@@ -0,0 +1,302 @@
+  useEffect(() => {
+    gradientRef.current = gradient;
+  }, [gradient]);
+
+  useEffect(() => {
+    let newGrad = colorsListRef.current
+      .map((color, index) => {
+        return `${color.color} ${color.position}%`;
+      })
+      .join(", ");
+
+    if (colorsListRef.current.length > 0) {
+      if (colorsListRef.current.length == 1) {
+        setGradient(`${colorsListRef.current[0].color}`);
+        gradientRef.current = `${colorsListRef.current[0].color}`;
+      } else {
+        if (gradientType.value === "linear-gradient") {
+          setGradient(
+            `${gradientType.value}(${gradientRotation.value}deg, ${newGrad})`
+          );
+          gradientRef.current = `${gradientType.value}(${gradientRotation.value}deg, ${newGrad})`;
+        } else if (gradientType.value === "conic-gradient") {
+          setGradient(
+            `${gradientType.value}(from ${gradientRotation.value}deg, ${newGrad})`
+          );
+          gradientRef.current = `${gradientType.value}(from ${gradientRotation.value}deg, ${newGrad})`;
+        } else {
+          setGradient(`${gradientType.value}(${newGrad})`);
+          gradientRef.current = `${gradientType.value}(${newGrad})`;
+        }
+      }
+    }
+  }, [colorsList, gradientType, gradientPosition, gradientRotation]);
+
+  const generateRandomGradient = () => {
+    function createHex() {
+      var hexCode1 = "";
+      var hexValues1 = "0123456789abcdef";
+
+      for (var i = 0; i < 6; i++) {
+        hexCode1 += hexValues1.charAt(
+          Math.floor(Math.random() * hexValues1.length)
+        );
+      }
+      return hexCode1;
+    }
+
+    var deg = Math.floor(Math.random() * 360);
+    const randomNoOfColors = Math.floor(Math.random() * 4) + 2; // generate random number of colors between 1 and 4
+    var gradient = "";
+    var colors = [];
+
+    for (var i = 0; i < randomNoOfColors; i++) {
+      colors.push({
+        id: `item_${i + 1}`,
+        color: "#" + createHex(),
+        position: i * 100,
+      });
+    }
+
+    colors.sort((a, b) => a.position - b.position);
+
+    colors.forEach((color, index) => {
+      gradient += `${color.color} ${color.position}%${
+        index < colors.length - 1 ? ", " : ""
+      }`;
+    });
+
+    setGradient(`linear-gradient(${deg}deg, ${gradient})`);
+    gradientRef.current = `linear-gradient(${deg}deg, ${gradient})`;
+    setColorsList(colors);
+    setGradientRotation({
+      id: gradientRotations.length + 1,
+      name: `${deg}°`,
+      value: deg,
+    });
+
+    // setGradient(gradient);
+    // gradientRef.current = gradient;
+    // setColorsList([
+    //   {
+    //     id: 1,
+    //     color: "#" + createHex(),
+    //     position: 0,
+    //   },
+    //   {
+    //     id: 2,
+    //     color: "#" + createHex(),
+    //     position: 100,
+    //   },
+    // ]);
+  };
+
+  return (
+    <main className="" class="bg-gray-900">
+      <title>Web dev tools</title>
+      <Nav></Nav>
```

Comment:
Please use the Nav that the rest of the pages use.
```suggestion
       <nav className="bg-blue-500 py-4 px-6 flex items-center justify-between h-15">
        <a
          href="https://web-dev-tools.vercel.app/"
          className="mr-2 flex border items-center rounded p-2 hover:bg-blue-600"
        >
          <h1 className="text-white text-lg md:text-2xl font-bold mr-4">
            Web Dev Tools
          </h1>
          <p>Gradient generator</p>
        </a>
        <Search />
      </nav>
```

Reply:
Ok